### PR TITLE
chore: remove `v2` naming related to legacy frontend

### PR DESCRIPTION
**Pull request by BugenZhao**
## What's changed and what's your intention?

This PR removes `v2` naming from the catalog, risedev, and crate names.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
- #2539
- #2555

### CODE REVIEW

File: proto/meta.proto [hunk 1]
```diff
@@ -262,9 +262,9 @@ message SubscribeResponse {
   uint64 version = 3;
   oneof info {
     common.WorkerNode node = 4;
-    catalog.Database database_v2 = 8;
-    catalog.Schema schema_v2 = 9;
-    catalog.Table table_v2 = 10;
+    catalog.Database database = 8;
```
Comment (yezizp2012) [inline proto/meta.proto:265]: Maybe also fix the sequence?

**Review by yezizp2012 — APPROVED** [review]
LGTM

**Review by skyzh — APPROVED** [review]

File: proto/meta.proto (hunk 1 above)
Comment (BugenZhao) [reply proto/meta.proto:265]: So we don't need to do forward compatibility yet? 🤣

Comment (yezizp2012) [reply proto/meta.proto]: No, not yet I guess. 🤣

**Pull request merged**